lib/Marlin/Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -293,23 +387,34 @@ void GcodeSuite::G28_no_parser(bool always_home_all, bool O, float R, bool S, bo
     #if DISABLED(DELTA) || ENABLED(DELTA_HOME_TO_SAFE_ZONE)
       const uint8_t old_tool_index = active_extruder;
     #endif
+    // PARKING_EXTRUDER homing requires different handling of movement / solenoid activation, depending on the side of homing
+    #if ENABLED(PARKING_EXTRUDER)
+      const bool pe_final_change_must_unpark = parking_extruder_unpark_after_homing(old_tool_index, X_HOME_DIR + 1 == old_tool_index * 2);
+    #endif
     tool_change(0, true);
   #endif
 
-  #if HAS_DUPLICATION_MODE
-    extruder_duplication_enabled = false;
-  #endif
+  TERN_(HAS_DUPLICATION_MODE, set_duplication_enabled(false));
 
   remember_feedrate_scaling_off();
 
   endstops.enable(true); // Enable endstops for next homing move
 
   #if ENABLED(DELTA)
 
+    constexpr bool doZ = true; // for NANODLP_Z_SYNC if your DLP is on a DELTA
+
     home_delta();
-    UNUSED(always_home_all);
 
-  #else // NOT DELTA
+    TERN_(IMPROVE_HOMING_RELIABILITY, end_slow_homing(saved_motion_state));
```

Comment:
This is supitious, as code above:
```
  #if ENABLED(IMPROVE_HOMING_RELIABILITY)
    slow_homing_t slow_homing{0};
    slow_homing.acceleration.set(planner.settings.max_acceleration_mm_per_s2[X_AXIS],
                                 planner.settings.max_acceleration_mm_per_s2[Y_AXIS]);
    planner.settings.max_acceleration_mm_per_s2[X_AXIS] = XY_HOMING_ACCELERATION;
    planner.settings.max_acceleration_mm_per_s2[Y_AXIS] = XY_HOMING_ACCELERATION;
    #if HAS_CLASSIC_JERK
      slow_homing.jerk_xy = planner.max_jerk;
      planner.max_jerk.set(XY_HOMING_JERK, XY_HOMING_JERK);
    #endif
    planner.reset_acceleration_rates();
  #endif
```
was not altered by this commit. I am not sure this change is compatible with the mentioned code above.

Reply:
IMPROVE_HOMING_RELIABILITY is currently disabled, and it breaks with this PR as I'm keeping this just about G28 to reduce the amount of changes to review. I preserved the changes to it in this case.

I already completed IMPROVE_HOMING_RELIABILITY but this spans planner/motion, will be submitted in another PR.

Reply:
IMPROVE_HOMING_RELIABILITY is enabled on mk4. It was enabled just in mk4 branch but today when we started new mk4 release we noticed this unmerged and is merged now.

Reply:
I wasn't aware. In this case, I added an additional commit that completes IMPROVE_HOMING_RELIABILITY as a whole.